tests/unit/plugins/modules/test_ec2_ami_info.py

```diff
@@ -0,0 +1,253 @@
+
+
+def test_get_images():
+    ec2_client = MagicMock()
+    module = MagicMock()
+
+    ec2_client.describe_images.return_value = {
+        "Images": [
+            {
+                "Architecture": "x86_64",
+                "BlockDeviceMappings": [
+                    {
+                        "DeviceName": "/dev/sda1",
+                        "Ebs": {
+                            "DeleteOnTermination": "True",
+                            "Encrypted": "False",
+                            "SnapshotId": "snap-0f00cba784af62428",
+                            "VolumeSize": 10,
+                            "VolumeType": "gp2"
+                        }
+                    }
+                ],
+                "ImageId": "ami-1234567890",
+                "ImageLocation": "1234567890/test-ami-uefi-boot",
+                "ImageType": "machine",
+                "Name": "test-ami-uefi-boot",
+                "OwnerId": "1234567890",
+                "PlatformDetails": "Linux/UNIX",
+            }
+        ],
+    }
+
+    request_args = {'ImageIds': ['ami-1234567890']}
+
+    get_images_result = ec2_ami_info.get_images(ec2_client, module, request_args)
+
+    ec2_client.describe_images.call_count == 2
+    ec2_client.describe_images.assert_called_with(aws_retry=True, **request_args)
+    assert len(get_images_result['Images']) == 1
+
+
+def test_get_image_attribute():
+    ec2_client = MagicMock()
+
+    ec2_client.describe_image_attribute.return_value = {
+        "ImageId": "ami-1234567890",
+        "LaunchPermissions": [
+            {
+                "UserId": "1234567890"
+            },
+            {
+                "UserId": "0987654321"
+            }
+        ]
+    }
+
+    image = {
+        "architecture": "x86_64",
+        "blockDeviceMappings": [
+            {
+                "device_name": "/dev/sda1",
+                "ebs": {
+                    "delete_on_termination": "True",
+                    "encrypted": "False",
+                    "snapshot_id": "snap-0f00cba784af62428",
+                    "volume_size": 10,
+                    "volume_Type": "gp2"
+                }
+            }
+        ],
+        "image_id": "ami-1234567890",
+        "image_location": "1234567890/test-ami-uefi-boot",
+        "image_type": "machine",
+        "name": "test-ami-uefi-boot",
+        "owner_id": "1234567890",
+        "platform_details": "Linux/UNIX"
+    }
+
+    get_image_attribute_result = ec2_ami_info.get_image_attribute(ec2_client, image)
+
+    ec2_client.describe_image_attribute.call_count == 1
+    ec2_client.describe_image_attribute.assert_called_with(aws_retry=True, Attribute='launchPermission', ImageId=image['image_id'])
+    assert len(get_image_attribute_result['LaunchPermissions']) == 2
+
+
+@patch(module_name + '.get_image_attribute')
+@patch(module_name + '.get_images')
+def test_list_ec2_images(m_get_images, m_get_image_attribute):
+    ec2_client = MagicMock()
```

Comment:
as this is repeated many times, you can create a fixture

Reply:
It is used only in `test_list_ec2_images`, I feel fixture could be unnecessary.

Reply:
Added.